Projects/Simba/simbaunit.pas

```diff
@@ -3551,17 +3549,29 @@ procedure TSimbaForm.ButtonPickClick(Sender: TObject);
   if SimbaSettings.ColourPicker.ShowHistoryOnPick.GetDefValue(True) then
     ColourHistoryForm.Show;
 
+  colour := inttostr(c);
   coordinate := inttostr(x) + ', ' + inttostr(y);
-  FormWritelnEx('Picked colour: ' + inttostr(c) + ' at (' + coordinate + ')');
+  FormWritelnEx('Picked colour: ' + colour + ' at (' + coordinate + ')');
+  if (SimbaSettings.ColourPicker.AddColourToClipBoard.GetDefValue(False)) then
+  begin
+    clipboardtext := colour;
+    if (SimbaSettings.ColourPicker.AddCoordinateToClipBoard.GetDefValue(False)) then
+      clipboardtext := clipboardtext + ', ' + coordinate;
+  end else
+    if (SimbaSettings.ColourPicker.AddCoordinateToClipBoard.GetDefValue(False)) then
+      clipboardtext := coordinate;
+
+  if clipboardtext = '' then
+    Exit;
+
   try
```

Comment:
Are you referring to 
```
if clipboardtext = '' then
  Exit;
```
This is a equality check, if both AddColourToClipBoard and AddCoordinateToClipBoard are false (which is the default), clipboardtext will be empty, so instead of setting our clipboard to empty we exit